sdk/src/azure/core/az_mqtt5_rpc_client_codec.c

```diff
@@ -58,10 +60,15 @@ AZ_NODISCARD az_result az_mqtt5_rpc_client_codec_get_publish_topic(
       : az_span_size(client->_internal.model_id) - az_span_size(az_mqtt5_rpc_service_id_key);
   if (command_id_index != -1)
   {
-    required_length += (az_span_is_content_equal(command_name, AZ_SPAN_EMPTY))
-        ? az_span_size(client->_internal.command_name)
-        : az_span_size(command_name);
-    required_length -= az_span_size(az_mqtt5_rpc_command_id_key);
+    if (az_span_is_content_equal(command_name, AZ_SPAN_EMPTY))
```

Comment:
```suggestion
    if (_az_span_is_valid(command_name, 1, false))
```
nit: just to catch if the size is zero but the pointer isn't NULL